### DISCUSSION TITLE
Annotations seem not to be accessible to screen readers

### DISCUSSION THREAD

**Discussion by nvdaes** [edrlab/thorium-reader#3039] [Developer corner]
@danielweb, I searched for keyboard commands in the default json file to find out how to create annotations. I've pressed NVDA+enter with NVDA, like I do to  create bookmarks. Also I have tried to select text in focus mode and browse mode, pressing shift+end, changing several settings from the annotations options button, and pressing alt gr + shift+a and alt gr +shift+q without results.
This is an important feature. If this is accessible, I don't find the documentation about it.
Thanks for this amazing work.

**Comment by danielweck** [comment]
Hello Noelia, I am @danielweck not @danielweb (smile)
The initial prerequisite before creating an annotation is to select some text in the HTML document. As usual with screen readers, this is more difficult than it is for sighted users who just use the mouse to drag a selection of characters inside the document. With NVDA I guess there is a mode that makes it possible to create real DOM selections.
Once this is done, then there is currently SHIFT CONTROL ALT key modifiers with A to invoke the equivalent of the button in the toolbar, this should open the annotations editor.
Please note that this is not yet quite mature enough for screen readers yet, due to the change of GUI components library we are now using in Thorium v3.
But please send us early feedback and we will keep on crushing those UX bugs, one by one!

**Comment by nvdaes** [comment]
Hi Daniel, sorry for my mistake typing your username (@danielweck ).
Unfortunately, NVDA doesn't have a mode to select text in the DOM for Chromium, just for Firefox:

https://github.com/nvaccess/nvda/pull/15830

The reason, quoting the explanation provided by @michaelDCurran , isthe following:

> Chromium implemented IAccessibleTextSelectionContainer, however much of the time its offset calculation seems to be wrong, thus the wrong native selection is made, thus this cannot be supported in Chromium at this point in time.

Perhaps on solution maybe to provide a multiline edit box to bookmarks in Thorium, to simulate the annotations feature. I did this in the placeMarkers add-on, thinking in epub readers that doesn't support this for screen readers. The add-on is available from NVDA store.
BTW, in Thorium 3.0 alphas, when I press control+b to add a bookmark, NVDA says alert, but not bookmark added. I think that this is a regression in Thorium.

**Comment by danielweck** [comment]
NVDA issue: https://github.com/nvaccess/nvda/issues/17591
